contracts/staking/Middleware.sol

```diff
@@ -0,0 +1,229 @@
+
+    function configureVault(address _vault, SlasherType _type) external onlyRole(SLASH_TYPE_SET_ROLE) {
+        require(_vault != address(0), "W:CV-Vault cannot be zero address");
+        
+        address collateral = IVault(_vault).collateral();
+        require(collateral != address(0), "W:CV-Collateral cannot be zero address");
+        // NOTE: slasher can be set later in vault, hence it can be zero address
+        address slasher = IVault(_vault).slasher();
+        require((slasher == address(0)) == (_type == SlasherType(0)), "W:CV-Slasher and type mismatch");
+        vaultInfo[_vault] = VaultInfo(uint80(vaults.length), _type, collateral, slasher);
+        vaults.push(_vault);
+
+        emit VaultInfoUpdated(_vault, collateral, _type);
+    }
+
+    function updateVault(address _vault, SlasherType _type) external onlyRole(SLASH_TYPE_SET_ROLE) {
+        require(_vault != address(0), "W:UV-Vault cannot be zero address");
+        require(vaultInfo[_vault].slasherType != SlasherType(0), "W:UV-Slasher not set");
+        require(vaultInfo[_vault].slasher == address(0), "W:UV-Slasher already set");
+
+        address slasher = IVault(_vault).slasher();
+        require(slasher != address(0), "W:UV-Collateral cannot be zero address");
+        vaultInfo[_vault].slasher = slasher;
+        vaultInfo[_vault].slasherType = _type;
+
+        emit VaultInfoUpdated(_vault, vaultInfo[_vault].collateral, _type);
+    }
+
+    function updateNetworkId(bytes32 _networkId) external onlyRole(NETWORK_UPDATE_ROLE) {
+        _updateNetworkId(_networkId);
+    }
+
+    function _updateNetworkId(bytes32 _networkId) internal {
+        networkId = _networkId;
+        emit NetworkIdUpdated(_networkId);
+    }
+    
+//-------------------------------- Slashing config end --------------------------------//
+
+//-------------------------------- Instant Slashing start --------------------------------//
+
+    function slash(
+        uint256 _jobId,
+        address _rewardAddress,
+        address _vault,
+        address _operator,
+        uint256 _amount,
+        uint48 _captureTimestamp,
+        bytes calldata _hints,
+        bytes calldata _proof
+    ) external {
+        require(vaultInfo[_vault].slasherType == SlasherType.INSTANT_SLASH, "M:S-Invalid slasher type");
+        require(_amount != 0, "M:S-Invalid amount");
+        require(slashInfo[_vault][_jobId].amount == 0, "M:S-Already slashed");
+
+        _verifyProof(abi.encode(_jobId, _rewardAddress, _vault, _operator, _amount, _captureTimestamp), _proof);
+
+        slashInfo[_vault][_jobId] = SlashInfo(_operator, _amount, _captureTimestamp, _rewardAddress);
+
+        IInstantSlasher(vaultInfo[_vault].slasher).slash(networkId, _operator, _amount, _captureTimestamp, _hints);
+
+        emit SlashProposed(_jobId, _vault, _operator, _amount, _captureTimestamp, _rewardAddress);
+    }
```

Comment:
Maybe batchSlash? as there could be multiple request?
Not sure if they still should be submitted one by one by the slash transmitter hmm

Reply:
right, we could add batch slash as well. Let's add it to the list of tasks for next iteration since changes are needed on slashing bridge code as well.